packages/pipeline/src/scripts/pull_radar_orderbook_snapshots.ts

```diff
@@ -0,0 +1,56 @@
+import { logUtils } from '@0x/utils';
+import { RadarMarket } from '@radarrelay/types';
+import * as R from 'ramda';
+import { Connection, ConnectionOptions, createConnection } from 'typeorm';
+
+import { RADAR_SOURCE, RadarSource } from '../data_sources/radar';
+import { TokenOrderbookSnapshot as TokenOrder } from '../entities';
+import * as ormConfig from '../ormconfig';
+import { parseRadarOrders } from '../parsers/radar_orders';
+import { handleError } from '../utils';
+
+// Number of orders to save at once.
+const BATCH_SAVE_SIZE = 1000;
+
+// Number of markets to retrieve orderbooks for at once.
+const MARKET_ORDERBOOK_REQUEST_BATCH_SIZE = 50;
+
+// Delay between market orderbook requests.
+const MILLISEC_MARKET_ORDERBOOK_REQUEST_DELAY = 5000;
+
+let connection: Connection;
+
+(async () => {
+    connection = await createConnection(ormConfig as ConnectionOptions);
+    const radarSource = new RadarSource();
+    const markets = await radarSource.getActiveMarketsAsync();
+    for (const marketsChunk of R.splitEvery(MARKET_ORDERBOOK_REQUEST_BATCH_SIZE, markets)) {
+        await Promise.all(
+            marketsChunk.map(async (market: RadarMarket) => getAndSaveMarketOrderbookAsync(radarSource, market)),
+        );
+        await new Promise<void>(resolve => setTimeout(resolve, MILLISEC_MARKET_ORDERBOOK_REQUEST_DELAY));
+    }
+    process.exit(0);
+})().catch(handleError);
+
+/**
+ * Retrieve orderbook from radar API for a given market. Parse orders and insert
+ * them into our database.
+ * @param radarSource Data source which can query radar API.
+ * @param market Object from radar API containing market data.
+ */
+async function getAndSaveMarketOrderbookAsync(radarSource: RadarSource, market: RadarMarket): Promise<void> {
+    const orderBook = await radarSource.getMarketOrderbookAsync(market.id);
+    const observedTimestamp = Date.now();
```

Comment:
My instinct is that we should use the same `observedTimestamp` for all markets, so we can use it as a task identifier.

Reply:
It's too late for this as all other DEX scraping jobs do this -- which also makes sense because orderbooks change pretty quickly in some cases.